src/modules/modal/modal.jsx

```diff
@@ -0,0 +1,132 @@
+    return new Error(
+      `One of props 'aria-label' or 'aria-labelledby' isRequired in '${componentName}'.`
+    )
+  }
+
+  if (!props['aria-label'] && !props['aria-labelledby']) {
+    return new Error(
+      `One of props 'aria-label' or 'aria-labelledby' was not specified in '${componentName}'.`
+    )
+  }
+
+  if (props['aria-label'] && props['aria-labelledby']) {
+    return new Error(
+      `Props 'aria-label' and 'aria-labelledby' in '${componentName}' mutually exclusive.`
+    )
+  }
+
+  return true
+}
+
+const Modal = ({
+  children,
+  onExit,
+  className,
+  hideManually = false,
+  selector = 'body',
+  ...restProps
+}) => {
+  const modalRef = useRef(null)
+
+  const handleKeyDown = useCallback(
+    (event) => {
+      if (hideManually || !onExit) return
+      if (event.key === 'Escape' || event.key === 'Esc' || event.keyCode === 27)
+        onExit(event)
+    },
+    [hideManually]
+  )
+
+  const handleClick = useCallback(
+    (event) => {
+      if (hideManually || !onExit) return
+      if (
+        (modalRef.current && modalRef.current.contains(event.target)) ||
+        // If the click is on the scrollbar we don't want to close the modal.
+        event.pageX > event.target.ownerDocument.documentElement.offsetWidth ||
+        event.pageY > event.target.ownerDocument.documentElement.offsetHeight
+      )
+        return
+      onExit(event)
+    },
+    [hideManually]
+  )
+
+  return (
+    <ModalPortal selector={selector}>
+      <ScrollLock>
+        <FocusLock returnFocus>
+          {/* eslint-disable-next-line max-len */}
+          {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions */}
+          <div
+            className={styles.overlay}
+            onClick={handleClick}
+            onKeyDown={handleKeyDown}
+            {...restProps}
+          >
+            <div
+              ref={modalRef}
+              role="dialog"
+              className={classNames.use(styles.modal).join(className)}
+              {...restProps}
+            >
+              {children}
+            </div>
+          </div>
+        </FocusLock>
+      </ScrollLock>
+    </ModalPortal>
+  )
+}
+
+Modal.Title = ModalTitle
+Modal.Content = ModalContent
+Modal.Footer = ModalFooter
+
+Modal.propTypes = {
+  'id': PropTypes.string,
+  'children': PropTypes.node,
+  /**
+   *  Callback called when modal should be closed
+   */
+  'onExit': PropTypes.func,
+  /**
+   *  Describes the title of the modal
+   */
+  'aria-label': checkAccessibility,
+  /**
+   * Points to modal label.
+   * Ideally it should be used in conjunction with Modal.Title
+   */
+  'aria-labelledby': checkAccessibility,
+  /**
+   * If set to true onExit callback is not triggered on overlay click
+   * or hitting ESC button.
+   */
+  'hideManually': PropTypes.bool,
+  /**
+   *  The location where the modal is rendered
+   */
+  'selector': PropTypes.string,
+  /**
+   * className applied on modal
+   */
+  'className': PropTypes.string,
```

Comment:
I thought we have validation disabled for the `className`, don't we? If not, I guess we should disable validation for the `className` and `id` unless we want to require any of these props.

Reply:
We don't.

Why do you want to disable validation for it?

Reply:
Because these attributes are global and obvious from the HTML which we base on. Having validation for it is an overhead making us write more, have longer code and the documentation but repeating what is already known.

Reply:
I see it's reported here https://github.com/oacore/design/issues/51. So once the issue is solved it can be removed. For the time being, I will keep it here.